src/main/java/org/cristalise/kernel/entity/ItemImplementation.java

```diff
@@ -109,31 +115,58 @@ public void initialise(SystemKey agentId, String propString, String initWfString
             PropertyArrayList props = (PropertyArrayList) Gateway.getMarshaller().unmarshall(propString);
             for (Property thisProp : props.list) mStorage.put(mItemPath, thisProp, locker);
         }
-        catch (Throwable ex) {
+        catch (Exception ex) {
             Logger.msg(8, "ItemImplementation::initialise(" + mItemPath + ") - Properties were invalid: " + propString);
             Logger.error(ex);
             mStorage.abort(locker);
             throw new InvalidDataException("Properties were invalid");
         }
 
-        // Store an event and the initial properties
+        History hist = new History(mItemPath, locker);
+
+        // Store an event and the outcome for initial properties
         try {
             Schema initSchema = LocalObjectLoader.getSchema("ItemInitialization", 0);
             Outcome initOutcome = new Outcome(0, propString, initSchema);
-            History hist = new History(mItemPath, locker);
+
             Event newEvent = hist.addEvent(new AgentPath(agentId), null, "", "Initialize", "", "", initSchema, Bootstrap.getPredefSM(), PredefinedStep.DONE, "last");
+
             initOutcome.setID(newEvent.getID());
             Viewpoint newLastView = new Viewpoint(mItemPath, initSchema, "last", newEvent.getID());
             mStorage.put(mItemPath, initOutcome, locker);
             mStorage.put(mItemPath, newLastView, locker);
         }
-        catch (Throwable ex) {
+        catch (Exception ex) {
             Logger.msg(8, "ItemImplementation::initialise(" + mItemPath + ") - Could not store event and outcome.");
             Logger.error(ex);
             mStorage.abort(locker);
             throw new PersistencyException("Error storing event and outcome");
         }
 
+        // Store an event and the outcome
+        if (StringUtils.isNotBlank(initViewpointString))
+            try {
+                Viewpoint vp = (Viewpoint)Gateway.getMarshaller().unmarshall(initViewpointString);
+                Schema schema = LocalObjectLoader.getSchema(vp.getSchemaName(), vp.getSchemaVersion());
+                Outcome outcome = new Outcome(-1, initOutcomeString, schema);
+                outcome.validateAndCheck();
+
+                vp.setItemPath(mItemPath);
+
+                Event newEvent = hist.addEvent(new AgentPath(agentId), null, "", "Initialize", "", "", schema, Bootstrap.getPredefSM(), PredefinedStep.DONE, vp.getName());

```

Comment:
Could we have 2 different events for property and details initialization? I'm thinking we might need to distinguish between them when displaying the history to the users.